src/main/java/life/qbic/model/download/QbicDataDownloader.java

```diff
@@ -184,12 +177,10 @@ private void downloadFile(DataSetFile dataSetFile, Path prefix) throws IOExcepti
       InputStream initialStream = file.getInputStream();
       CheckedInputStream checkedInputStream = new CheckedInputStream(initialStream, new CRC32());
       if (file.getDataSetFile().getFileLength() > 0) {
-        final Path filePath = determineFinalPathFromDataset(file.getDataSetFile());
-        File newFile =
-                new File(System.getProperty("user.dir") +
-                        File.separator +
-                        prefix.toString() + File.separator +
-                        filePath.toString());
+        final Path filePath = OutputPathFinder.determineFinalPathFromDataset(file.getDataSetFile(), conservePaths);
+        final Path finalPath = OutputPathFinder.determineOutputDirectory(outputPath, prefix, file.getDataSetFile(), conservePaths);
```

Comment:
NIT: This could be more expressive
```suggestion
        final Path folderPath = OutputPathFinder.determineOutputDirectory(outputPath, prefix, file.getDataSetFile(), conservePaths);
```